plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoIpProcessor.java

```diff
@@ -68,13 +71,21 @@ public void execute(Data data) {
             throw new RuntimeException(e);
         }
 
-        final Map<String, Object> geoData;
+        Map<String, Object> geoData;
         switch (dbReader.getMetadata().getDatabaseType()) {
             case "GeoLite2-City":
-                geoData = retrieveCityGeoData(ipAddress);
+                try {
+                    geoData = retrieveCityGeoData(ipAddress);
+                } catch (AddressNotFoundRuntimeException e) {
+                    geoData = Collections.emptyMap();
+                }
                 break;
             case "GeoLite2-Country":
-                geoData = retrieveCountryGeoData(ipAddress);
+                try {
+                    geoData = retrieveCountryGeoData(ipAddress);
+                } catch (AddressNotFoundRuntimeException e) {
+                    geoData = Collections.emptyMap();
```

Comment:
sure, maybe a separate discussion but I guess we will need to log something sooner or later. anyways we can discuss this later on. let's get this in first